libparsec/crates/client/tests/unit/certif/add_device_certificate.rs

```diff
@@ -27,6 +27,27 @@ async fn ok(env: &TestbedEnv) {
     p_assert_matches!(switch, MaybeRedactedSwitch::NoSwitch);
 }
 
+#[parsec_test(testbed = "minimal")]
+async fn multiple(env: &TestbedEnv) {
+    let env = env.customize(|builder| {
+        builder.new_device("alice");
+    });
+    let alice = env.local_device("alice@dev1");
+    let ops = certificates_ops_factory(&env, &alice).await;
+
+    let switch = ops
+        .add_certificates_batch(
+            &env.get_common_certificates_signed(),
```

Comment:
Given the test is about adding multiple certificates, it would be good to have an assert about the number of certificates:
```rust
let to_add_certifs = env.get_common_certificates_signed();
p_assert_eq!(to_add_certif.len(), 2);  // Sanity check
...
let switch = ops
        .add_certificates_batch(
            &to_add_certifs,
...
```

Reply:
Don't forget that there is also all the required certificates, and even if you want to fetch and add your own certificates, the `env` will create `alice user & device`

Reply:
yes, my example is broken here, I should have done something like:

```rust
let all_common_certifs = env.get_common_certificates_signed();
let to_add_certifs = all_common_certifs[all_common_certifs.len() - 2..];
```
Or even better: have the builder return the list of certificates that have been created since the `builder.certificates_storage_fetch_certificates()`

Reply:
Need to add this sanity check for others tests then

Reply:
Of course the `len` is `2` if you keep only `2` :joy:
```Rust
let to_add_certifs = all_common_certifs[all_common_certifs.len() - 2..];
```